### PR TITLE
Fixes for tests for Windows

### PR DESCRIPTION
**Pull request by tiemevanveen**
Tests were failing on windows. 

The app did not seem to start, looks like this issue: http://stackoverflow.com/questions/37343150/spectron-and-electron-without-exe-files

Eventually, the timeout will fire without even the `before` to be finished. 

This PR fixes that. 

I've also addressed the `userDataPath` since it was only configured for OSX. 

Tests are still failing on Windows btw, but that seems unrelated. This line evaluates to `false` when I run the tests.

      .browserWindow.isFocused().should.eventually.be.true

https://github.com/electron/electron-api-demos/blob/master/tests/index.js#L113

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -18,7 +19,24 @@ describe('demo app', function () {
...
     const productName = require('../package').productName
-    const userDataPath = path.join(process.env.HOME, 'Library', 'Application Support', productName)
+
+    let userDataPath = ''
+    switch (process.platform) {
+      case 'darwin':
+        userDataPath = path.join(process.env.HOME, 'Library', 'Application Support', productName)
+        break
+      case 'win32':
+        userDataPath = path.join('%APPDATA%', productName)
```
Comment (tiemevanveen) [inline tests/index.js]: Hmm, it seems `%APPDATA%` won't work. `process.env.HOME, 'AppData', 'Roaming'` might, but it depends a bit on where the app is loaded from right? Any suggestions?

Comment (tiemevanveen) [inline tests/index.js]: Should have been `process.env.APPDATA` :)

**Comment by zeke** [comment]
Thanks for catching this, @tiemevanveen. There's actually a method built in to Electron for getting this path: [app.getPath('appData')](http://electron.atom.io/docs/api/app#appgetpathname)

**Comment by tiemevanveen** [comment]
Yeah I know, but you need to boot the app first to use that API right? Or would there be a way to use that before booting the app?

**Comment by zeke** [comment]
Ah, I see. There seem to be lots of modules that abstract this away. [ospath](https://github.com/jprichardson/ospath#readme) might be a good candidate.

**Comment by tiemevanveen** [comment]
Check. Would you prefer taking in another dependency over this? There is really no way to use the method from Electron right? That's a pity. 

I tried to lookup the code / dependency that they are usig but can't find it in de source.

**Comment by zeke** [comment]
> I tried to lookup the code 

Yeah I did too a bit. Looks like the configuration is set in [brightray](https://github.com/electron/brightray).

@kevinsawicki any thoughts on this?

**Comment by kevinsawicki** [comment]
> There is really no way to use the method from Electron right?

Little hacky, but you could spawn Electron and have it print out the `userData` path:

```js
const {app} = require('electron')

app.once('ready', function () {
  console.log(app.getPath('userData'))
})
```

You can specify the above script as a preload script on the command line by giving its path like:

```sh
electron -r ./print-user-data-path.js .
```

**Comment by tiemevanveen** [comment]
Wel at least it will execute the same code to get the path.. but a little hacky indeed. 

And what about hardcoding the paths, but adding a test that asserts that they're equal once the app is booted and we've got access to the runtime?

**Comment by kevinsawicki** [comment]
> And what about hardcoding the paths, but adding a test that asserts that they're equal once the app is booted and we've got access to the runtime?

Sounds fine to me 👍

**Comment by zeke** [comment]
Hi @tiemevanveen. It sounds like "hardcoding the paths, but adding a test that asserts that they're equal once the app is booted" is an approach we can all live with. Do you need any helping pushing this over the finish line?

**Comment by tiemevanveen** [comment]
> Do you need any helping pushing this over the finish line?

Cool. Yeah I'll probably need some help once the PR is done. I'll work on this tomorrow. Thanks for the reminder.

**Comment by tiemevanveen** [comment]
Added the test, but please check cause I can't imagine this is the best way.. 

```
return require('electron').remote.require('electron').app.getPath('userData')
```

But couldn't make it work another way. Not a lot experience with electron, sorry.

**Comment by zeke** [comment]
Does it work if you take out that second `require` statement?

```js
return require('electron').remote.app.getPath('userData')
```

**Comment by tiemevanveen** [comment]
> Does it work if you take out that second require statement?

Yeah that works as well. Of course. Thanks :)

**Comment by zeke** [comment]
Awesome work, @tiemevanveen! Thanks for your perseverance.

**Pull request merged**